hubparkour-api/src/main/java/me/block2block/hubparkour/api/events/player/ParkourPlayerTeleportEvent.java

```diff
@@ -5,25 +5,77 @@
 import me.block2block.hubparkour.api.plates.PressurePlate;
 
 /**
- * Called when a user uses /parkour reset or /parkour checkpoint.
+ * Called when a user uses /parkour reset or /parkour checkpoint or player fall.
  */
 @SuppressWarnings("ALL")
 public class ParkourPlayerTeleportEvent extends ParkourPlayerEvent {
 
     private final PressurePlate plate;
+    private final TeleportReason teleportReason;
 
     @SuppressWarnings("unused")
-    public ParkourPlayerTeleportEvent(IParkour parkour, IHubParkourPlayer player, PressurePlate pressurePlate) {
+    public ParkourPlayerTeleportEvent(IParkour parkour, IHubParkourPlayer player,
+                                      PressurePlate pressurePlate, TeleportReason teleportReason) {
         super(parkour, player);
         this.plate = pressurePlate;
+        this.teleportReason = teleportReason;
     }
 
     /**
      * Get the pressure plate that the user teleported to.
+     *
      * @return the pressure plate
      */
     @SuppressWarnings("unused")
     public PressurePlate getPressurePlate() {
         return plate;
     }
+
+    /**
+     * Get reason of player teleport
+     *
+     * @return the reason of teleport
+     */
+    @SuppressWarnings("unused")
+    public TeleportReason getTeleportReason() {
+        return teleportReason;
+    }
+
+    /**
+     * The reason of teleport
+     */
+    public enum TeleportReason {
+        /**
+         * Unknow reason
+         */
+        UNKNOW,
```

Comment:
You still have a typo here, should be
```suggestion
        UNKNOWN,
```